libraries/SITL/SIM_GPS.cpp

```diff
@@ -289,7 +292,7 @@ void GPS::update_ubx(const struct gps_data *d)
         uint32_t headVeh;
         uint8_t reserved2[4]; 
     } pvt {};
-    const uint8_t SV_COUNT = 10;
+    const uint8_t SV_COUNT = 13;
```

Comment:
Unmotivated change?

Your PR title says "Trimble GSOF GPS Simulator", so I really shouldn't be seeing patches around ubx

Likewise NMEA and probably others I haven't spotted.  I mean, these look like good changes, but...

Reply:
I can revert it, however this just seemed wrong. The autotest suite tries to configure the number of simulated GPS's to 13, but this hard codes a limit of 10. I can split it off.

Reply:
Also, as far as the other changes, it felt wrong to keep copying the same lines of code to compute heading and ground speed(extra flash and code maintenance).

There are other places like getting the number of satellites that had different behavior in each driver.

Reply:
This specific change is now moved out here: https://github.com/ArduPilot/ardupilot/pull/24523